pkg/build/builder/sti.go

```diff
@@ -239,6 +239,21 @@ func (s *S2IBuilder) Build() error {
 		}
 	}
 
+	if config.Incremental {
+		if s.build.Spec.Strategy.SourceStrategy.ForcePull || !isImagePresent(s.dockerClient, config.IncrementalFromTag) {
+			startTime := metav1.Now()
+			err = s.pullImage(config.IncrementalFromTag, t)
+			timing.RecordNewStep(ctx, buildapiv1.StagePullImages, buildapiv1.StepPullInputImage, startTime, metav1.Now())
+			// If there was an error, the incremental image may not exist. Treat the build as a normal s2i build.
+			if err != nil {
```

Comment:
Can we not check what kind of error was returned and process based on that?

Reply:
We may be able to check the error equals `ErrNoSuchImage` (this comes from `fsouza/go-dockerclient`)

Reply:
the `s.pullImage` logic also has a retry loop.